web/core/components/issues/attachment/attachment-list-item.tsx

```diff
@@ -0,0 +1,111 @@
+"use client";
+
+import { FC } from "react";
+import { observer } from "mobx-react";
+import { Trash } from "lucide-react";
+// ui
+import { CustomMenu, Tooltip } from "@plane/ui";
+// components
+import { ButtonAvatars } from "@/components/dropdowns/member/avatar";
+import { getFileIcon } from "@/components/icons";
+import { IssueAttachmentDeleteModal } from "@/components/issues";
+// helpers
+import { convertBytesToSize, getFileExtension, getFileName } from "@/helpers/attachment.helper";
+import { renderFormattedDate } from "@/helpers/date-time.helper";
+// hooks
+import { useIssueDetail, useMember } from "@/hooks/store";
+import { usePlatformOS } from "@/hooks/use-platform-os";
+// types
+import { TAttachmentOperations } from "./root";
+
+type TAttachmentOperationsRemoveModal = Exclude<TAttachmentOperations, "create">;
+
+type TIssueAttachmentsListItem = {
+  attachmentId: string;
+  handleAttachmentOperations: TAttachmentOperationsRemoveModal;
+  disabled?: boolean;
+};
+
+export const IssueAttachmentsListItem: FC<TIssueAttachmentsListItem> = observer((props) => {
+  // props
+  const { attachmentId, handleAttachmentOperations, disabled } = props;
+  // store hooks
+  const { getUserDetails } = useMember();
+  const {
+    attachment: { getAttachmentById },
+    isDeleteAttachmentModalOpen,
+    toggleDeleteAttachmentModal,
+  } = useIssueDetail();
+  // derived values
+  const attachment = attachmentId ? getAttachmentById(attachmentId) : undefined;
+  // hooks
+  const { isMobile } = usePlatformOS();
+
+  if (!attachment) return <></>;
+
+  return (
+    <>
+      {isDeleteAttachmentModalOpen === attachment.id && (
```

Comment:
We don't need to put this information in store.

Reply:
Resolved